File: test/ruby/DataTypeListString.rb
#!/usr/bin/env ruby
require 'hyperdex'

def assert &block
    raise RuntimeError unless yield
end

c = HyperDex::Client::Client.new(ARGV[0], ARGV[1].to_i)
assert { c.put("kv", "k", {}) == true }
assert { c.get("kv", "k") == {:v => []} }
assert { c.put("kv", "k", {"v" => ['A', 'B', 'C']}) == true }
assert { c.get("kv", "k") == {:v => ['A', 'B', 'C']} }
assert { c.put("kv", "k", {"v" => []}) == true }
assert { c.get("kv", "k") == {:v => []} }

File: test/python/DataTypeListString.py
#!/usr/bin/env python
import sys
import hyperdex.client
from hyperdex.client import LessEqual, GreaterEqual, LessThan, GreaterThan, Range, Regex, LengthEquals, LengthLessEqual, LengthGreaterEqual
c = hyperdex.client.Client(sys.argv[1], int(sys.argv[2]))
def to_objectset(xs):
    return set([frozenset(x.items()) for x in xs])
assert c.put('kv', 'k', {}) == True
assert c.get('kv', 'k') == {'v': []}
assert c.put('kv', 'k', {'v': ['A', 'B', 'C']}) == True
assert c.get('kv', 'k') == {'v': ['A', 'B', 'C']}
assert c.put('kv', 'k', {'v': []}) == True
assert c.get('kv', 'k') == {'v': []}
